hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetTypeManagement.java

```diff
@@ -139,6 +144,29 @@ public void delete(final Long targetTypeId) {
         return convertPage(targetTypeRepository.findAll(spec, pageable), pageable);
     }
 
+    @Override
+    public Page<TargetType> findByTargetTypeFilter(Pageable pageable, TargetTypeFilter targetTypeFilter) {
+
+        final List<Specification<JpaTargetType>> specList = Lists.newArrayListWithExpectedSize(2);
+
+        Specification<JpaTargetType> spec;
+
+        if (!StringUtils.isEmpty(targetTypeFilter.getSearchText())) {
+            spec = TargetTypeSpecification.likeNameOrDescription(targetTypeFilter.getSearchText());
+            specList.add(spec);
+        }
+
+         if (!StringUtils.isEmpty(targetTypeFilter.getFilterString())) {
+            spec = TargetTypeSpecification.likeNameOrDescription(targetTypeFilter.getFilterString().trim() + "%");
```

Comment:
I think it's confusing when you also find TargetTypes that have the String in its description when searching for it in the combobox. So we should only search the name for the String